root/layout/components/Search.js

```diff
@@ -13,24 +13,26 @@ import * as manifest from '../../static/manifest';
 import * as DBDefs from '../../static/scripts/common/DBDefs';
 import {l, lp} from '../../static/scripts/common/i18n';
 
+/* eslint-disable sort-keys, flowtype/sort-keys */
 const TYPE_OPTIONS = {
-  artist:         l('Artist'),
-  release_group:  l('Release Group'),
-  release:        l('Release'),
-  recording:      l('Recording'),
-  work:           l('Work'),
-  label:          l('Label'),
-  area:           l('Area'),
-  place:          l('Place'),
-  annotation:     l('Annotation'),
-  cdstub:         l('CD Stub'),
-  editor:         l('Editor'),
-  tag:            lp('Tag', 'noun'),
-  instrument:     l('Instrument'),
-  series:         lp('Series', 'singular'),
-  event:          l('Event'),
-  doc:            DBDefs.GOOGLE_CUSTOM_SEARCH ? l('Documentation') : null,
+  artist:        l('Artist'),
```

Comment:
The iteration order for an object is technically arbitrary and undefined, it just so happens that most engines respect the insertion order. I wouldn't rely on it though (as an anecdote, a lot of our Perl code broke when we updated to 5.18 for the same reason). We should be using a [Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map).

Reply:
Agreed, I replaced it with an array of option groups, and, within each group, sorted options with localized `compare` on titles in https://github.com/metabrainz/musicbrainz-server/pull/639/commits/bb2532d239eb845befa8eca1527ec340f9013619.